### ISSUE TITLE
De-duplicate output

### ISSUE THREAD

**Issue by ioggstream** [italia/api-oas-checker#79]
## Reproduce

- reference multiple times an errored component in the spec, eg

```
x-foo: &xrll
  Test:
    $ref: '#/components/headers/Test'
components:
  headers:
    Test:
      description: ciao
      schema:
        type: integer
  responses:
    400BadRequest:
      description: ciao
      headers:
       <<: *xrll
    404NotFound:
      description: ciao
      headers:
       <<: *xrll
paths: {}
openapi: 3.0.0
```

## I expect

- errors to be reported once

```
line 2: number-boundaries
```

## Instead

- errors are reported multiple times.

```
line 2: number-boundaries
line 2: number-boundaries
line 2: number-boundaries
```

## Note

The problem is related to the *expansion* of entries (eg. the referenced schema is the same,
but paths are different.

```
#/components/responses/400BadRequest/headers/Test/schema 
                                          ^^^^^^vvvvvvvv
#/components/responses/404NotFound/headers/Test/schema 
```

We have a similar issue when referencing schemas

```

components:
  headers:
    Test:
      description: ciao
      schema:
        type: integer
  responses:
    400BadRequest:
      description: ciao
      headers:
        Test:
          $ref: '#/components/headers/Test'
    404NotFound:
      description: ciao
      headers:
        Test:
          $ref: '#/components/headers/Test'
paths: {}
openapi: 3.0.0
```

**Comment by ioggstream** [comment]
@bfabio a possible fix is not as simple as I thought :)

**Comment by emilioSp** [comment]
@ioggstream should the _export results_ capability return _raw validation results_ or can we use the deduplicated results?

**Comment by ioggstream** [comment]
The raw one, to retain compatibility with Spectral.
Dedupe should be a feature of the UI.
